test/util/e2e.go

```diff
@@ -232,3 +231,22 @@ func RestartKueueController(ctx context.Context, k8sClient client.Client) {
 	kcmKey := types.NamespacedName{Namespace: "kueue-system", Name: "kueue-controller-manager"}
 	rolloutOperatorDeployment(ctx, k8sClient, kcmKey)
 }
+
+func TerminatePod(ctx context.Context, k8sClient client.Client, restClient *rest.RESTClient, cfg *rest.Config, name, namespace string, exitCode int) {
+	pod := corev1.Pod{}
+	podKey := client.ObjectKey{Namespace: namespace, Name: name}
+	gomega.Eventually(func(g gomega.Gomega) string {
+		g.Expect(k8sClient.Get(ctx, podKey, &pod)).To(gomega.Succeed())
+		return pod.Status.PodIP
+	}, Timeout, Interval).ShouldNot(gomega.BeEmpty())
+
+	cmd := []string{"/bin/sh", "-c", fmt.Sprintf("curl \"http://%s:8080/exit?code=%v&timeout=1s&wait=0s\"", pod.Status.PodIP, exitCode)}
+	_, _, err := KExecute(ctx, cfg, restClient, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name, cmd)
+	if err != nil {
+		exitErr, ok := err.(*exec.ExitError)
+		// sometimes pod gets killed by completed job causes the request to fail with 137
```

Comment:
this code and comment are cryptic. Certainly the command does not care about status of the job. The command might be indeed failing with 137 if the pod is already terminated, but
- we should minimize the risk by only calling the stop for pods which aren't stopped yet

I think we could introduce a new function WaitForActivePodsAndTerminate(active pods count, exitCode) which will wait for the expected number of active pods and then terminate them. Also, I think waiting and terminating should be done in different Eventually.

I think then we don't need special handling for 137. Or if there is a special test requiring it still then we can pass a flag to do so from the test.

It will also help to save on code duplication as currently the code is repeated in a couple of tests.